dev-packages/ext-scripts/theia-ext.js

```diff
@@ -48,19 +58,26 @@ function run(script) {
 }
 
 (async () => {
+    /** @type {Error | number} */
     let exitCode = 0;
     let extScript = undefined;
     try {
         extScript = getExtScript();
+        console.debug(`$ ${extScript}`);
         exitCode = await run(extScript);
     } catch (err) {
         if (extScript) {
-            console.error(`Error occurred in theiaext when executing: '${extScript}'`, err);
+            console.error(`Error occurred in theiaext when executing: ${extScript}\n`);
         } else {
-            console.error('Error occurred in theiaext', err);
+            console.error('Error occurred in theiaext.');
         }
-        console.log(`${err.name}: ${err.message}`);
-        exitCode = 1;
+        console.error(err);
+        exitCode = err;
```

Comment:
Should not exitCode be a number?

Reply:
Yes, but later on I check if it is not a number and convert it before calling `process.exit`.

Because when `child_process` throws an error, it often means the sub-process didn't even start, so there isn't any exit code to report either. So I store the error and then report `exitCode = 1`.